ntoskrnl/mm/section.c

```diff
@@ -4863,9 +4783,6 @@ MmCheckDirtySegment(
         Entry = WRITE_SSE(Entry);
         MmSetPageEntrySectionSegment(Segment, Offset, Entry);
 
-        /* Tell the other users that we are clean again */
-        MmSetCleanAllRmaps(Page);
```

Comment:
Can you please describe what (was|is) going on here?

Reply:
This was supposed to remove the dirty bit of all PTEs mapping this page. While this theorically works, all this did was potentially harmful because it was impossible to get proper locking.
Now the worst case is to flush the same data more than once to disk, which is still better than a deadlock.